File: 05/CSharp/ExpressionApp/Multiply.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExpressionApp
{
    class Multiply : BinaryOperator
    {
        public Multiply(IExpression first, IExpression second)
        {
            fOp = first;
            sOp = second;
        }

        public override double Evaluate(double param)
        {
            return fOp.Evaluate(param) * sOp.Evaluate(param);
        }
    }
}

File: 05/Java/expression/Multiply.java
package Expression;

public class Multiply extends BinaryOperator implements Expression {
    public Multiply(Expression first, Expression second) {
        fOp = first;
        sOp = second;
    }

    public double evaluate(double param) {
        return fOp.evaluate(param) * sOp.evaluate(param);
    }
}